src/dreyfus_httpd.erl

```diff
@@ -418,6 +426,74 @@ parse_non_negative_int_param(Name, Val, Prop, Default) ->
     end.
 
 
+validate_search_restrictions(Db, DDoc, Args) ->
+    #index_query_args{
+        q = Query,
+        partition = Partition,
+        grouping = Grouping,
+        limit = Limit
+    } = Args,
+    #grouping{
+        by = GroupBy
+    } = Grouping,
+
+    case Query of
+        undefined ->
+            Msg1 = <<"Query must include a 'q' or 'query' argument">>,
+            throw({query_parse_error, Msg1});
+        _ ->
+            ok
+    end,
+
+    DbPartitioned = fabric_util:is_partitioned(Db),
+    ViewPartitioned = get_view_partition_option(DDoc, DbPartitioned),
+
+    case not DbPartitioned andalso is_binary(Partition) of
+        true ->
+            Msg2 = <<"`partition` not supported on this index">>,
+            throw({bad_request, Msg2});
+        false ->
+            ok
+    end,
+
+    case {ViewPartitioned, is_binary(Partition)} of
+        {false, false} ->
+            ok;
+        {true, true} ->
+            ok;
+        {true, false} ->
```

Comment:
iiuc, this means this design doc has a partitioned view, which means this design doc is part of a partitioned db, and therefore our search index must also be partitioned. Perhaps we can write `Msg3` better. It might confuse users doing a text partition search incorrectly but getting a view errror. Maybe something along the lines of:
`partition parameter is mandatory for this index`

Reply:
Done.